nifty-ssl/src/main/java/com/facebook/nifty/ssl/SslConfigFileWatcher.java

```diff
@@ -0,0 +1,146 @@
+ * limitations under the License.
+ */
+package com.facebook.nifty.ssl;
+
+import com.facebook.nifty.core.NettyServerTransport;
+import com.google.common.collect.ImmutableSet;
+import io.airlift.log.Logger;
+import org.apache.tomcat.jni.SessionTicketKey;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Watches SSL config files for changes and updates the {@link SslServerConfiguration} of the attached
+ * {@link NettyServerTransport} when the files change. Three kind of files are watched:
+ * <ul>
+ * <li>a ticket seed file. New ticket seeds are constructed from it with a {@link TicketSeedFileParser}.</li>
+ * <li>a TLS private key file. Passed to the SslServerConfiguration, which loads the key from the file.</li>
+ * <li>a TLS certificate file. Passed to the SslServerConfiguration, which loads the cert from the file.</li>
+ * </ul>
+ * Note that the file paths cannot be changed after the watcher is created. The watcher will automatically start
+ * polling files for changes when it is attached to a NettyServerTransport, and will stop polling when it is
+ * detached from the transport. All three file paths are currently required.
+ */
+public class SslConfigFileWatcher implements TransportAttachObserver {
+    private final AtomicReference<NettyServerTransport> transportRef;
+    private final File ticketSeedFile;
+    private final File keyFile;
+    private final File certFile;
+    private final TicketSeedFileParser ticketSeedFileParser;
+    private final MultiFileWatcher watcher;
+
+    private static final Logger log = Logger.get(PollingMultiFileWatcher.class);
+
+    /**
+     * Constructs a new config file watcher.
+     *
+     * @param ticketSeedFile the path to the ticket seed file. May not be null.
+     * @param keyFile the path to the TLS private key file. May not be null.
+     * @param certFile the path to the TLS certificate file. May not be null.
+     * @param ticketSeedSalt the ticket seed salt. If null, uses {@link TicketSeedFileParser#DEFAULT_TICKET_SALT}.
+     * @param watcher the {@link MultiFileWatcher} to use for monitoring SSL config files for changes.
+     */
+    public SslConfigFileWatcher(File ticketSeedFile,
+                                File keyFile,
+                                File certFile,
+                                byte[] ticketSeedSalt,
+                                MultiFileWatcher watcher) {
+        transportRef = new AtomicReference<>(null);
+        this.ticketSeedFile = requireNonNull(ticketSeedFile);
+        this.keyFile = requireNonNull(keyFile);
+        this.certFile = requireNonNull(certFile);
+        ticketSeedFileParser = new TicketSeedFileParser(ticketSeedSalt);
+        this.watcher = requireNonNull(watcher);
+    }
+
+    @Override
+    public void attachTransport(NettyServerTransport transport) {
```

Comment:
attach/detach don't seem to be used/tested anywhere?

also why would you do this at all if transport is null? seems like you should just requireNonNull and get rid of null handling code